src/main/java/com/dreamsportslabs/guardian/service/AuthorizeService.java

```
@@ -0,0 +1,149 @@
+      String loginChallenge,
+      AuthorizeSessionModel sessionModel,
+      AuthorizeRequestDto requestDto,
+      String tenantId,
+      String loginPageUri,
+      int authorizeTtl) {
+
+    return authorizeSessionDao.saveAuthorizeSession(
+        loginChallenge, sessionModel, tenantId, authorizeTtl);
+  }
+
+  private ClientModel validateClient(
+      ClientModel client, AuthorizeRequestDto requestDto, String tenantId) {
+    validateRedirectUri(client, requestDto);
+    validateResponseType(client, requestDto);
+    validateScope(requestDto);
+    return client;
+  }
+
+  private void validateRedirectUri(ClientModel client, AuthorizeRequestDto requestDto) {
+    if (client.getRedirectUris() == null
+        || !client.getRedirectUris().contains(requestDto.getRedirectUri())) {
+      throw INVALID_REDIRECT_URI.getJsonCustomException("Redirect uri is invalid");
+    }
+  }
+
+  private void validateResponseType(ClientModel client, AuthorizeRequestDto requestDto) {
+    if (client.getResponseTypes() == null
+        || !client.getResponseTypes().contains(requestDto.getResponseType())) {
+      throw UNSUPPORTED_RESPONSE_TYPE.getRedirectCustomException(
+          "Unsupported response_type", requestDto.getState(), requestDto.getRedirectUri());
+    }
+  }
+
+  private void validateScope(AuthorizeRequestDto requestDto) {
+    String[] scopes = requestDto.getScope().split("\\s+");
+    for (String scope : scopes) {
+      if (SCOPE_OPENID.equals(scope.trim())) {
+        return;
+      }
+    }
+    throw INVALID_SCOPE.getRedirectCustomException(
+        "scope must contain 'openid'", requestDto.getState(), requestDto.getRedirectUri());
+  }
+
+  private List<String> filterAllowedScopes(
+      String requestedScopes, List<ClientScopeModel> clientScopes) {
+    Set<String> clientAllowedScopes =
+        clientScopes.stream().map(ClientScopeModel::getScope).collect(Collectors.toSet());
+
+    return Arrays.stream(requestedScopes.split("\\s+"))
```

Should we throw an error when there are invalid scopes?

---

Spec defines to ignore invalid scopes